### DISCUSSION TITLE
Implement Kaiser waveform ?

### DISCUSSION THREAD

**Discussion by LaurentAjdnik** [pasqal-io/Pulser#227] [Ideas] [answered]
Hi all!

From https://arxiv.org/abs/2104.15044:

`In practice, a properly shaped waveform—like the Blackman waveform, which we will use here — can mitigate unwanted modifications to the modulated signal, stemming from spectral leakage or noise sensitivity.`

From https://numpy.org/doc/stable/reference/generated/numpy.blackman.html:

`The Blackman window is a taper formed by using the first three terms of a summation of cosines. It was designed to have close to the minimal leakage possible. It is close to optimal, only slightly worse than a Kaiser window.`

Hence, should we implement a `KaiserWaveform` based on https://numpy.org/doc/stable/reference/generated/numpy.kaiser.html (probably with `beta = 8.6`)?

**Comment by HGSilveri** [comment, marked as answer]
Hey @LaurentAjdnik ! This a very good idea, I'm all for it. 

To be clear, the `beta` should be a keyword argument, and the default at `beta=8.6`.

Can I turn this into an issue and assign it to you?

**Reply by HGSilveri** [reply]
On second thought, you might want to start with `beta=14`, as suggested in the numpy docs. Otherwise, we'll just have this one duplicating the `BlackmanWaveform` by default.

**Reply by LaurentAjdnik** [reply]
Yup!

This is either gonna be super simple or a real pain in the *ss... :grin:

**Comment by LaurentAjdnik** [comment]
Hi @HGSilveri, you can mark this as "Answered".
